recipes/cli/all/conanfile.py

```diff
@@ -0,0 +1,81 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMakeToolchain, CMake, CMakeDeps
+from conan.tools.files import get, copy, rmdir
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+import os
+
+class CLIConan(ConanFile):
+    name = "cli"
+    description = "A library for interactive command line interfaces in modern C++"
+    license = "BSL-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/daniele77/cli"
+    topics = "cli-interface", "cpp14", "no-dependencies", "cli", "header-only"
+    package_type = "header-library"
+    settings = "os", "compiler", "build_type", "arch"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    # in case the project requires C++14/17/20/... the minimum compiler version should be listed
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "6",
+            "Visual Studio": "16",
+            "msvc": "192",
+            "apple-clang": "14",
+        }
+
+    def package_id(self):
+        self.info.clear()
+
+    def validate(self):
+        # validate the minimum cpp standard supported. Only for C++ projects
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
```

Comment:
If the recipe does not declare any requirements, this can be safely skipped.

Nevertheless, I see in the upstream CMakeLists that it has some options for using `asio` that were skipped here. Any insight into why you decided to omit them? I don't oppose it if it makes contributing the recipe easier, it can be left as an exercise for the next PR if someone needs it, but happy to hear your thoughts :)

Reply:
Since this is a header only library it is left to the user of the library if they want to include the CLI_UseBoostAsio flag when compiling their project.  If we added it as an option here then we would need to ensure this flag was used in any consuming code for the option to make sense.  I am not familiar with how that would work but if you have an example we can give it a try.

Reply:
@RubenRBS Let me know if you think this should be changed or if you have any comments based on my comment.  I believe everything else is addressed.

Reply:
Hello @rob-baily Looking into https://github.com/daniele77/cli/blob/master/CMakeLists.txt#L70 it seems like you have both Boost and asio as requirement. I'll send you a PR with a suggestion to include them as requirements.